test/unittest/tests/block/test_block_file.cpp

```diff
@@ -0,0 +1,214 @@
+}
+
+void
+validate_block(WT_BLOCK *block, std::map<std::string, std::string> &config_map, uint expected_ref,
+  std::string name, bool readonly = false, bool created_during_backup = false)
+{
+    REQUIRE(block != nullptr);
+
+    // Test Block immediate members.
+    REQUIRE(std::string(block->name).compare(name) == 0);
+    REQUIRE(block->objectid == WT_TIERED_OBJECTID_NONE);
+    // REQUIRE(block->compact_session_id = WT_SESSION_ID_INVALID);
+    REQUIRE(block->ref == expected_ref);
+    REQUIRE(block->readonly == readonly);
+    REQUIRE(block->created_during_backup == created_during_backup);
+    REQUIRE(block->extend_len == 0);
+
+    // Test Block file handle members.
+    validate_block_fh(block, expected_ref, name);
+    REQUIRE(std::string(block->live_lock.name).compare("block manager") == 0);
+    REQUIRE(block->live_lock.initialized == true);
+
+    // Test Block configuration members.
+    validate_block_config(block, config_map);
+}
+
+void
+validate_and_free_block(WT_SESSION_IMPL *session, WT_BLOCK *block, config_parser &cp,
+  uint expected_ref, std::string name, bool readonly = false, bool created_during_backup = false)
+{
+    validate_block(block, cp.get_config_map(), expected_ref, name, readonly, created_during_backup);
+    free_block(session, block);
+}
+
+TEST_CASE("Block: __wt_block_open", "[block_file]")
+{
+    /* Build Mock session, this will automatically create a mock connection. */
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    config_parser cp({{"allocation_size", ALLOCATION_SIZE}, {"block_allocation", BLOCK_ALLOCATION},
+      {"os_cache_max", OS_CACHE_MAX}, {"os_cache_dirty_max", OS_CACHE_DIRTY_MAX},
+      {"access_pattern_hint", ACCESS_PATTERN}});
+
+    REQUIRE((session->getMockConnection()->setupBlockManager(session->getWtSessionImpl())) == 0);
+
+    SECTION("Normal case")
+    {
+        WT_BLOCK *block;
+        REQUIRE(
+          (__wt_block_open(session->getWtSessionImpl(), DEFAULT_FILE_NAME.c_str(),
+            WT_TIERED_OBJECTID_NONE, cp.get_config_array(), false, false, false, 0, &block)) == 0);
+        validate_block(block, cp.get_config_map(), 1, DEFAULT_FILE_NAME);
+
+        // Test already made item in hashmap.
+        WT_BLOCK *block2;
+        REQUIRE(
+          (__wt_block_open(session->getWtSessionImpl(), DEFAULT_FILE_NAME.c_str(),
+            WT_TIERED_OBJECTID_NONE, cp.get_config_array(), false, false, false, 0, &block2)) == 0);
+        validate_and_free_block(session->getWtSessionImpl(), block2, cp, 2, DEFAULT_FILE_NAME);
+        validate_and_free_block(session->getWtSessionImpl(), block, cp, 1, DEFAULT_FILE_NAME);
+        /*
+         * Test already made item in hashmap but with different configuration.
+         * WT_BLOCK *block3;
+         * cp.get_config_map()["allocation_size"] = "1024";
+         * REQUIRE((__wt_block_open(session->getWtSessionImpl(), DEFAULT_FILE_NAME.c_str(),
```

Comment:
Currently if there is an block created with the same name but different configuration. We would return the block with the old configuration. I wonder if this is a bug or expected behaviour?

Reply:
Hi Jie, that's a great question. I don't know the answer right now, but the fact that creating the test has simply raised the question is very valuable.

Reply:
To me that sounds like expected behaviour as I'd assume names are meant to be unique, it sounds like something we should create a ticket to document and not something that needs fixing.

Reply:
I wonder if we should re-organise this so we can assert that names are unique? If a different allocation size comes in, we assert that it is not possible. What do you think @luke-pearson ?

Reply:
We assert that in the block manager code? We can do that, create a ticket 👍

Reply:
Created documentation ticket [WT-13503](https://jira.mongodb.org/browse/WT-13503)